File: internal/content/testdata/bench/javascript/majority-element-half.js
function majority_element(nums) {
    let count = 0;
    let candidate = null;
    for (const n of nums) {
        if (count === 0) candidate = n;
        count += (n === candidate) ? 1 : -1;
    }
    const freq = nums.filter(x => x === candidate).length;
    if (freq > Math.floor(nums.length / 2)) return candidate;
    return -1;
}

File: internal/content/testdata/bench/java/majority-element-half.java
import java.util.*;

class Solution {
    public long majority_element(long[] nums) {
        long count = 0, candidate = 0;
        for (long n : nums) {
            if (count == 0) candidate = n;
            count += (n == candidate) ? 1 : -1;
        }
        long freq = 0;
        for (long n : nums) if (n == candidate) freq++;
        if (freq > nums.length / 2) return candidate;
        return -1;
    }
}
